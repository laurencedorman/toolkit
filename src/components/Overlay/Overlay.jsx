/**
 * Copyright (c) Colibri SAS - ManoMano
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { animated, Transition } from 'react-spring';
import styles from './Overlay.module.scss';

const Overlay = ({ onClick, on, zIndex, backgroundColor }) => {
  const classNames = cn('toolkit', styles.overlay);

  return (
    <Transition
      native
      config={{ precision: 0.9 }}
      items={on}
      from={{ o: 0 }}
      enter={{ o: 0.5 }}
      leave={{ o: 0 }}
    >
      {display =>
        display &&
        // eslint-disable-next-line react/prop-types,react/display-name
        (({ o }) => (
          <animated.div
            className={classNames}
            onClick={onClick}
            style={{
              // eslint-disable-next-line react/prop-types
              opacity: o.interpolate(opacity => opacity),
              cursor: onClick && 'pointer',
              zIndex,
              backgroundColor,
            }}
          />
        ))
      }
    </Transition>
  );
};

Overlay.displayName = 'Overlay';

Overlay.propTypes = {
  onClick: PropTypes.func,
  on: PropTypes.bool.isRequired,
  zIndex: PropTypes.number,
  backgroundColor: PropTypes.string,
};

Overlay.defaultProps = {
  zIndex: 8000,
  backgroundColor: '#000',
  onClick: null,
};

export default Overlay;
