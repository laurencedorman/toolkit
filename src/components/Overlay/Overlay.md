Warning

- It includes react-spring as dependency (31.79KB)

Overlay :

```js
import Button from '../Button';
import Toggle from '../Toggle';

<Toggle>
  {({ on, toggle }) => (
    <div>
      <Button onClick={toggle}>Open overlay</Button>
      <Overlay onClick={toggle} on={on} />
    </div>
  )}
</Toggle>;
```
