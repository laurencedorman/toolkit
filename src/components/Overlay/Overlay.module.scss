/**
 * Copyright (c) Colibri SAS - ManoMano
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

@import '../../styles/import.scss';

.overlay {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: $black;
  z-index: z($layers, fullscreen);
  will-change: opacity;
}
