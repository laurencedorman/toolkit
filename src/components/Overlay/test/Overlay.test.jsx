/**
 * Copyright (c) Colibri SAS - ManoMano
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { mount } from 'enzyme';
import Overlay from '../Overlay';

describe('Overlay', () => {
  let wrapper;

  const props = {
    zIndex: 8000,
    backgroundColor: '#000',
    onClick: jest.fn(),
  };

  wrapper = mount(<Overlay on={false} {...props} />);

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render overlay if on is true', () => {
    wrapper.setProps({ on: true });
    expect(wrapper.props().on).toBe(true);
  });

  it('should have pointer cursor if component has props onClick', () => {
    wrapper = mount(<Overlay on {...props} />);
    expect(wrapper.find('div').prop('style')).toHaveProperty(
      'cursor',
      'pointer'
    );
  });
});
