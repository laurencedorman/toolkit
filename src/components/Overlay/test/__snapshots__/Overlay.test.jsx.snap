// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay should match snapshot 1`] = `
<Overlay
  backgroundColor="#000"
  on={false}
  onClick={[MockFunction]}
  zIndex={8000}
>
  <Transition
    config={
      Object {
        "precision": 0.9,
      }
    }
    enter={
      Object {
        "o": 0.5,
      }
    }
    from={
      Object {
        "o": 0,
      }
    }
    items={false}
    keys={[Function]}
    leave={
      Object {
        "o": 0,
      }
    }
    native={true}
    reset={false}
    unique={false}
  >
    <ForwardRef
      config={
        Object {
          "precision": 0.9,
        }
      }
      delay={0}
      filter={[Function]}
      from={
        Object {
          "o": 0,
        }
      }
      key="0"
      native={true}
      primitive={[Function]}
      reset={false}
      state="enter"
      states={
        Object {
          "enter": Object {
            "o": 0.5,
          },
        }
      }
    >
      <KeyframesImpl
        config={
          Object {
            "precision": 0.9,
          }
        }
        delay={0}
        filter={[Function]}
        forwardRef={null}
        from={
          Object {
            "o": 0,
          }
        }
        native={true}
        primitive={[Function]}
        reset={false}
        state="enter"
        states={
          Object {
            "enter": Object {
              "o": 0.5,
            },
          }
        }
      >
        <Spring
          config={
            Object {
              "precision": 0.9,
            }
          }
          delay={0}
          force={false}
          from={
            Object {
              "o": 0,
            }
          }
          immediate={false}
          inject={[Function]}
          native={true}
          onRest={[Function]}
          reset={false}
          to={
            Object {
              "o": 0.5,
            }
          }
        />
      </KeyframesImpl>
    </ForwardRef>
  </Transition>
</Overlay>
`;
